src/ol/renderer/webgl/Layer.js

```diff
@@ -84,6 +85,13 @@ class WebGLLayerRenderer extends LayerRenderer {
      * @protected
      */
     this.helper;
+
+    layer.addChangeListener(
+      LayerProperty.Z_INDEX,
+      this.removeHelper_.bind(this)
+    );
+
+    layer.addChangeListener(LayerProperty.MAP, this.removeHelper_.bind(this));
```

Comment:
Are these really needed? A change in the layer render order would either cause the cache key to change (and thus the helper to be removed) or stay the same, which means the helper can be kept as is. Same for the map id, a change of this property would cause the cache key to change.

Reply:
Right.  Initially, I only reset the helper on these events.  In writing tests, I realized the shortcoming of this approach since the grouping for other layers can change when one is removed or adjusted in z-index.

I left the listeners in since it makes it so removing a layer (and not re-adding it) results in disposing of the helper.  But the z-index listener is not strictly necessary any more.

Reply:
I removed the z-index listener in the latest commit.